Fix #17185: MusicXML ignores custom string data

HemantAntony opened the pull request:
Resolves: #17185 <!-- Replace `NNNNN` with a GitHub issue number, or a direct link if the issue is not on GitHub -->

<!-- Add a short description of and motivation for the changes here -->

<!-- Replace `[ ]` with `[x]` to fill the checkboxes below -->

- [x] I signed the [CLA](https://musescore.org/en/cla)
- [x] The title of the PR describes the problem it addresses
- [x] Each commit's message describes its purpose and effects, and references the issue it resolves
- [x] If changes are extensive, there is a sequence of easily reviewable commits
- [x] The code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [x] There are no unnecessary changes
- [x] The code compiles and runs on my machine, preferably after each commit individually
- [x] I created a unit test or vtest to verify the changes I made (if applicable)

HemantAntony commented:
This PR also solves another unreported problem. When you tune a string to a -1 octave, say C-1 and export it to MusicXML, the exported file gets corrupted because the `minInclusive` value of `tuning-octave` in the schema is 0

Jojo-Schmitz commented:
I don't quite understand how the changes of this PR could lead to those unit test failures, which are about the number of frets but not their tuning? But I guess just adjusting that test is the correct way of dealing with this

lvinken commented:
A quick check shows at least the staff tuning is imported correctly with this PR.

Please also add one or more test files.

DmitryArefiev commented:
#17185 also looks fine on my side (Windows10) , but tests should be fixed at first. @HemantAntony Can you look into it please?

Jojo-Schmitz commented:
That mtest failure is related to the 1st commit, not to the 2nd

HemantAntony commented:
Fixed the tests and updated the PR. I also added two tests to test both problems (#17185 and negative octave).

Also should I add PDFs of the test files? I see there are PDFs of `xml` and `mscx`

Jojo-Schmitz commented on src/importexport/musicxml/internal/musicxml/exportxml.cpp:
```diff
@@ -6775,6 +6775,19 @@ void ExportMusicXml::writeElement(EngravingItem* el, const Measure* m, staff_idx
+
+static void clampMusicXmlOctave(int& octave)
+{
+    octave = std::clamp(octave, 0, 9);
```
This seems to be C++17 and later only, am I right that `octave = std::max(std::min(octave, 9), 0);` is the same thing for older C++ standards?

DmitryArefiev commented:
Tested #17185 on Win10, Mac13 - **FIXED**    

@HemantAntony Thanks for fixing!
